### ISSUE TITLE
Server side authentication API functions

### ISSUE THREAD

**Issue by FuzzyStatic** [esnet/iperf#872]
# Enhancement Request

* Current behavior

As of iperf3.6, there are no libiperf APIs that allow for configuration of authentication for an iperf3 server process.

* Desired behavior

Add three new api calls to accompany server mode:

```
void    iperf_set_test_server_username(struct iperf_test *ipt, char *server_username);
void    iperf_set_test_server_password(struct iperf_test *ipt, char *server_password);
void    iperf_set_test_server_rsa_key(struct iperf_test *ipt, char *server_rsa_key_base64);
```

@ralcini could you implement this since you are the iperf3 authentication master? :)

**Comment by ralcini** [comment]
Hi @FuzzyStatic , do you need an API to submit an authenticated test request to a server on which is enabled authentication, right?

**Comment by FuzzyStatic** [comment]
Hey @ralcini, no, I'm looking for an API to setup authentication for a server.
So whatever credentials a client will have to use to authenticate to the server will be defined in this API. 

Basically API functions to do `iperf3 -s -p <port> --rsa-private-key-path <key> --authorized-users-path <username/password file>` but instead of having a file for the username and password it can be directly passed via the above functions.

Function `void    iperf_set_test_server_rsa_key(struct iperf_test *ipt, char *server_rsa_key_base64);` would define the private key on the server side. For clarity it could be `void    iperf_set_test_server_rsa_private_key(struct iperf_test *ipt, char *server_rsa_private_key_base64);`

**Comment by ralcini** [comment]
ok @FuzzyStatic , it's clear.... but: i think it's better to have an api that can be called multiple times to allow the same behaviour allowed by the command line. For example:

`iperf_add_test_server_credentials(struct iperf_test *ipt, char *server_username, char *server_password)`
and 
`iperf_reset_test_server_credentials(struct iperf_test *ipt)`

What do you think about it?

**Comment by FuzzyStatic** [comment]
@ralcini that's fine by me and I will defer to your judgement. As long as I can input the username, password, and private key I'm fine with either implementation.

**Comment by FuzzyStatic** [comment]
Hey @ralcini, have you had any time to look into this request?

**Comment by FuzzyStatic** [comment]
Any word of this?

**Comment by bmah888** [comment]
Fixed in #911 .
